modules/core/src/test/java/org/apache/ignite/internal/processors/cache/CacheDataRegionConfigurationTest.java

```diff
@@ -169,4 +173,157 @@ public void testProperlySizedMemoryPolicy() throws Exception {
             fail("With properly sized DataRegion no exceptions are expected to be thrown.");
         }
     }
+
+    /**
+     * Verifies that {@link IgniteCheckedException} is thrown when swap and persistence are enabled at the same time
+     * for a data region.
+     */
+    public void testSetPersistenceAndSwap() {
+        DataRegionConfiguration invCfg = new DataRegionConfiguration();
+
+        invCfg.setName("invCfg");
+        invCfg.setInitialSize(DFLT_MEM_PLC_SIZE);
+        invCfg.setMaxSize(DFLT_MEM_PLC_SIZE);
+        // Enabling the persistence.
+        invCfg.setPersistenceEnabled(true);
+        // Enabling the swap space.
+        invCfg.setSwapPath("/path/to/some/directory");
+
+        memCfg = new DataStorageConfiguration();
+        memCfg.setDataRegionConfigurations(invCfg);
+
+        ccfg = new CacheConfiguration(DEFAULT_CACHE_NAME);
+        ccfg.setDataRegionName("ccfg");
+
+        GridTestUtils.assertThrows(log(), new Callable<Object>() {
+            @Nullable @Override public Object call() throws Exception {
+                startGrid(0);
+                return null;
+            }
+        }, IgniteCheckedException.class, "Failed to start processor: GridProcessorAdapter []");
+    }
+
+    /**
+     * Verifies that {@link IgniteCheckedException} is thrown when page eviction threshold isn't between 0.5 and 0.999.
+     */
+    public void testSetInvalidEviction() {
```

Comment:
Ideally, additional test cases not related to the fix should be committed under a separate JIRA issue (like "Add more test cases to CacheDataRegionConfigurationTest"), but personally I'm OK with piggybacking them here as well.

Reply:
Got it.